pkg/cache/scheduler/tas_flavor_snapshot.go

```diff
@@ -1317,32 +1328,43 @@ func (s *TASFlavorSnapshot) fillInCounts(
 		domain.leaderState = 0
 	}
 	for _, leaf := range s.leaves {
-		// 1. Check Tolerations against Node Taints
-		taint, untolerated := corev1helpers.FindMatchingUntoleratedTaint(leaf.nodeTaints, tolerations, func(t *corev1.Taint) bool {
-			return t.Effect == corev1.TaintEffectNoSchedule || t.Effect == corev1.TaintEffectNoExecute
-		})
```

Comment:
This check was originally executed regardless of `isLowestLevelNode` and `leaf.nodeTaints` is only set when the last level is a leaf, so IIUC we can skip this if `!s.isLowestLevelNode()`, like the `selector` check.

I re-ordered the guard clauses because then this can be checked once + the original comments were invalid (comments said that `"2. Check Node Labels against Compiled Selector"` but it was actually done **third**, and `belongsToRequiredDomain` was second.